File: Week2 (Queue, Stack, Array)/Leetcode/0150-evaluate-reverse-polish-notation/cmy.py
def evalRPN(tokens):
    """
    :type tokens: List[str]
    :rtype: int
    """
    stack = []
    for token in tokens:
        try:
            stack.append(int(token))
        except:
            op2 = stack.pop()
            op1 = stack.pop()
            if token == "+":
                stack.append(op1+op2)
            elif token == "-":
                stack.append(op1-op2)
            elif token == "*":
                stack.append(op1*op2)
            else:
                stack.append(int(op1/op2))
    return stack.pop()

tokens = ["10","6","9","3","+","-11","*","/","*","17","+","5","+"]
print(evalRPN(tokens))

File: Week2 (Queue, Stack, Array)/Leetcode/0150-evaluate-reverse-polish-notation/cmy.c
#include <stdio.h>
#include <stdbool.h> // to use bool type
#include <ctype.h> // isdigit()
#include <stdlib.h> // atoi()

int evalRPN(char** tokens, int tokensSize) {
    int stack[tokensSize];
    int top = -1;
    for (int i=0; i<tokensSize; i++) {
        char* token = tokens[i];
        if (isdigit(token[0]) || (token[0] == '-' && isdigit(token[1]))) {
            stack[++top] = atoi(token);
        } else {
            int op2 = stack[top--];
            int op1 = stack[top--];
            switch (token[0]) {
                case '+': stack[++top] = op1 + op2; break;
                case '-': stack[++top] = op1 - op2; break;
                case '*': stack[++top] = op1 * op2; break;
                case '/': stack[++top] = op1 / op2; break;
            }
        }
    }
    return stack[top];
}

int main() {
    char* tokens[] = {"4", "13", "5", "/", "+"};
    printf("%d", evalRPN(tokens, 5));
    return 0;
}